pynest/nest/tests/test_tsodyks2_connection.py

```diff
@@ -0,0 +1,169 @@
+
+
+@nest.ll_api.check_stack
+class Tsodyks2ConnectionTest(unittest.TestCase):
+    """
+    Functional test for the "tsodyks2" synapse: compare NEST implementation to
+    a reference generated in the method reproduce_weight_drift(), for a
+    sequence of spike times coming from a Poisson generator.
+    """
+
+    def setUp(self):
+        self.resolution = 0.1  # [ms]
+        self.presynaptic_firing_rate = 20.0  # [Hz]
+        self.simulation_duration = 1E3  # [ms]
+        self.hardcoded_trains_length = 15.  # [ms]
+        self.synapse_parameters = {
+            "receptor_type": 1,
+            "delay": self.resolution,
+            "U": 1.0,
+            "u": 1.0,
+            "x": 1.0,
+            "tau_rec": 100.,
+            "tau_fac": 0.,
+            "weight": 1.            # initial weight
+        }
+
+    def test_tsodyk2_synapse(self):
+        synapse_model = "tsodyks2_synapse_rec"
+        self.synapse_parameters["synapse_model"] = synapse_model
+
+        pre_spikes, weight_by_nest = self.do_the_nest_simulation()
+        weight_reproduced_independently = self.reproduce_weight_drift(
+            pre_spikes,
+            self.synapse_parameters["weight"])
+
+        assert np.all(np.abs(np.array(weight_reproduced_independently) - np.array(weight_by_nest)) < 1E-12)
+
+    def do_the_nest_simulation(self):
+        """
+        This function is where calls to NEST reside.
+        Returns the generated pre- and post spike sequences
+        and the resulting weight established by the tsodyks2 synapse.
+        """
+        nest.set_verbosity('M_WARNING')
+        nest.ResetKernel()
+        nest.SetKernelStatus({'resolution': self.resolution})
+
+        neurons = nest.Create(
+            "parrot_neuron",
+            2,
+            params={})
+        presynaptic_neuron = neurons[0]
+        postsynaptic_neuron = neurons[1]
+
+        presynaptic_generator = nest.Create(
+            "poisson_generator",
+            1,
+            params={"rate": self.presynaptic_firing_rate,
+                    "stop": (self.simulation_duration - self.hardcoded_trains_length)})
+
+        # The detector is to save the randomly generated spike trains.
+        spike_detector = nest.Create("spike_detector")
+
+        nest.Connect(presynaptic_generator, presynaptic_neuron,
+                     syn_spec={"synapse_model": "static_synapse"})
+        nest.Connect(presynaptic_neuron + postsynaptic_neuron, spike_detector,
+                     syn_spec={"synapse_model": "static_synapse"})
+        # The synapse of interest itself
+        wr = nest.Create('weight_recorder', 1)
+        nest.CopyModel("tsodyks2_synapse", "tsodyks2_synapse_rec",
+                       {"weight_recorder": wr})
+        nest.Connect(presynaptic_neuron, postsynaptic_neuron,
+                     syn_spec=self.synapse_parameters)
+
+        nest.Simulate(self.simulation_duration)
+
+        all_spikes = nest.GetStatus(spike_detector, keys='events')[0]
```

Comment:
We can write this simpler in NEST 3.0:
```suggestion
        all_spikes = spike_detector.events
```